src/crimson/os/seastore/backref/backref_tree_node.h

```diff
@@ -67,10 +67,6 @@ class BackrefInternalNode
   extent_types_t get_type() const final {
     return TYPE;
   }
-
-  bool may_conflict() const final {
-    return false;
-  }
```

Comment:
I think we need to be really careful when trying to adjust behaviors related to conflicts.

This is to relax the constraint to allow conflicts on the backref-tree, meaning that this PR is allowing multiple transactions to update the backref-tree concurrently. However, I don't think that removing the `may_conflict()` just works. The current trim/clean transactions are not designed to be executed concurrently, one example is the weak transaction that bypasses the conflict mechanism during reclaim:

https://github.com/ceph/ceph/blob/c6786ea08e094f019fb75f11bc22c88edb9994d6/src/crimson/os/seastore/async_cleaner.cc#L1129-L1158 

There are also performance considerations referring to the discussions https://github.com/ceph/ceph/pull/47553#discussion_r944132824. I think the conclusion was to prevent the concurrency between trim and cleaning transactions.

Seems to me the above statement is also violated in this design, as the space reservation and transaction blocking is implemented during transaction submission, meaning that the trim transaction can only be blocked when it is already pending. Scheduling further cleaning transactions from the temp-cleaner introduces the violated concurrency.

A reasonable way to deal with the above issues is to introduce the size limit to `JournalTrimmerImpl::trim_dirty()`, so we are able to decide before creating trim transactions rather than blocking already running trim transactions. And in this way, we don't need to introduce major changes to the existing `BackgroundProcess`, for example:
```c++
future<> do_background_cycle() {
  // pseudo code for illustration
  if (trimmer->should_trim() && cleaner->try_reserve_space(trim-size-limit-per-cycle) {
    return trimmer->trim(
    ).safe_then([this] {
      cleaner->release_reserved_space(trim-size-limit-per-cycle);
    });
  } else if (cleaner->should_clean_space()) {
    return cleaner->clean_space();
  }
}
// minor adjustment to other places like `background_should_run()` are also needed.
```

Does it make sense?

Reply:
OK, I'll finish it in a few days